aten/src/ATen/TensorGeometry.h

```diff
@@ -27,35 +27,35 @@ struct TORCH_API TensorGeometry {
   }
 
   explicit TensorGeometry(const TensorBase& t)
-      : sizes_(t.sizes().vec()),
-        strides_(t.strides().vec()),
-        storage_offset_(t.storage_offset()),
-        numel_(t.numel()) {}
+      : sizes_(t.sym_sizes().vec()),
+        strides_(t.sym_strides().vec()),
+        storage_offset_(t.sym_storage_offset()),
+        numel_(t.sym_numel()) {}
 
   // true if the tensor is contiguous
   bool is_contiguous() const;
 
   int64_t dim() const {
     return sizes_.size();
   }
-  int64_t size(int64_t dim) const {
+  c10::SymInt size(int64_t dim) const {
     dim = c10::maybe_wrap_dim(dim, this->dim());
-    return sizes_.at(static_cast<size_t>(dim));
```

Comment:
removing the static cast is unnecessary and probably introduces a signed-unsigned conversion warning